src/foolscap/test/test_banana.py

```diff
@@ -1691,14 +1849,14 @@ def test_decimal(self):
         # check that new == old.
         d.addCallback(lambda res: self.loop(Decimal('NaN')))
         def _check_NaN(new_NaN):
-            self.failUnlessEqual(str(new_NaN), str(Decimal('NaN')))
+            self.assertEqual(str(new_NaN), str(Decimal('NaN')))
         d.addCallback(_check_NaN)
         return d
 
     def test_string(self):
         return self.looptest("biggles")
     def test_unicode(self):
-        return self.looptest(u"biggles\u1234")
+        return self.looptest("biggles\\u1234")
     def test_list(self):
```

Comment:
this will now be `str` and not `unicode` on py2, no? and the double-slash means no special unicode char?

Reply:
This was perhaps a wrong escape added by 2to3. Checking.

Reply:
Fixed.